### PR TITLE
[BLAS, cuBLAS backend, hipSYCL] Add official hipsycl cuBLAS support 

### PR DESCRIPTION
**Pull request by sbalint98**
As discussed under PR #144 hipSYCL rocBLAS support comes with automatic support for cuBLAS as well. This PR adds this information to the README. 

Furthermore, this PR differentiates the FindcuBLAS.cmake file between hipSYCL and Intel LLVM. In the case of hipSYCL, there is no need to have the `OPENCL_PATH` set. Moreover, the `CUDA_CUDA_LIBRARY` environment variable is also not strictly necessary since finding the libcuda.so is handled by hipSYCL.

[onemkl-cuda-hipsycl-test.txt](https://github.com/oneapi-src/oneMKL/files/8076543/onemkl-cuda-hipsycl-test.txt)

### CODE REVIEW

**Comment by mmeterel** [comment]
@sbalint98 As you are aware, there were several changes since the time this PR is opened. Could you please rebase and check if the tests are all good? Also, I think, we can enable cuRAND backend with hipSYCL with this PR as well. What do you think?

Could you please let me know when everything is ready?

**Comment by sbalint98** [comment]
Hi @mmeterel thank you for reminding me. I'll have a look and prepare this PR until Monday. I'll let you know if it is ready for review

**Comment by sbalint98** [comment]
Sorry for the delay. I have rebased the branch and updated the readme. All tests are passing on my sid
[cublas_tests_hipsycl.txt](https://github.com/oneapi-src/oneMKL/files/8846920/cublas_tests_hipsycl.txt)
e

**Comment by mmeterel** [comment]
> Sorry for the delay. I have rebased the branch and updated the readme. All tests are passing on my sid [cublas_tests_hipsycl.txt](https://github.com/oneapi-src/oneMKL/files/8846920/cublas_tests_hipsycl.txt) e

@sbalint98 Thanks. Do you plan to add cuRAND support in this PR as well?

**Comment by sbalint98** [comment]
@nilsfriess  and @normallytangent are working on the cuRAND support. I believe for them it might be easier to open a new PR. Would this be acceptable for you?

**Comment by mmeterel** [comment]
> @nilsfriess and @normallytangent are working on the cuRAND support. I believe for them it might be easier to open a new PR. Would this be acceptable for you?

Sure, that works if it will be easier on your side.

**Comment by mmeterel** [comment]
@sbalint98 
Thanks for the PR. I tested it locally (had to rebase hipSYCL and build it with LLVM-13) and it works smoothly. :)

**Review by mmeterel — APPROVED** [review]

**Comment by mmeterel** [comment]
@mkrainiuk Could you please take a look at this?

File: cmake/FindcuBLAS.cmake [hunk 1]
```diff
@@ -32,20 +32,36 @@ add_compile_definitions(CUDA_NO_HALF)
...
-
+  if(USE_ADD_SYCL_TO_TARGET_INTEGRATION)
+    find_package_handle_standard_args(cuBLAS
+        REQUIRED_VARS
+          CUDA_TOOLKIT_INCLUDE
+          CUDA_cublas_LIBRARY
+          CUDA_LIBRARIES
+          CUDA_CUDART_LIBRARY
+    )
+    set_target_properties(ONEMKL::cuBLAS::cuBLAS PROPERTIES
```
Comment (mkrainiuk) [inline cmake/FindcuBLAS.cmake:47]: Is there any reasons we cannot use one `set_target_properties`? I see the only difference: `OPENCL_INCLUDE_DIR` that can be added with inline check, something like "$<$<BOOL:${USE_ADD_SYCL_TO_TARGET_INTEGRATION}>: ${OPENCL_INCLUDE_DIR}>"

**Review by mkrainiuk — APPROVED** [review]
Thank you for the PR! Changes look good to me. I have one minor comment.

File: cmake/FindcuBLAS.cmake (hunk 1 above)
Comment (mmeterel) [reply cmake/FindcuBLAS.cmake:47]: Not sure...but i also see differences between CUDA_CUDART_LIBRARY vs CUDA_CUDA_LIBRARY

**Pull request merged**